UKHomeOffice/nmsw-ui
# NMSW-268 - Voyage FAL1 General Declaration upload

# Ticket

NMSW-268

----

## AC

Given a user has not uploaded a file
When they click check for errors
Then a form field styled error is shown

Given a user has uploaded a file
When a user clicks check for errors
Then the FE sends a POST to the /declaration/{declarationId}/upload-fal1 endpoint

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal1 endpoint
When the response is a 200
Then the FE shows the no errors found page

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal1 endpoint
And the file is not of type .csv or .xlsx
When they click check for errors
Then a form field styled 'invalid type' error is shown

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal1 endpoint
And the file is missing from the payload
Then a form field styled 'no file' error is shown

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal1 endpoint
And the endpoint is missing a declaration id
Then the something went wrong page is shown
And the link on that page returns the user to the `Your voyages` page

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal1 endpoint
And their token has expired/become invalid
Then the sign in page is shown
When they sign in
Then they are returned to the `upload General Declaration FAL 1` page

Given the FE has sent a POST to the /declaration/{declarationId}/upload-fal1 endpoint
And the API returns a 500 error
Then the something went wrong page is shown
And the link on that page returns the user to the  `General Declaration FAL 1` page


----

## To test

ERROR SCENARIOS:

- Log in
- Click on `Report a voyage` button
- > Upload general declaration FAL 1 page should display as per prototype
- Click on `FAL 1 general declaration` link
- > template should download
- Click on `Check for errors`
- > Error of “Select a FAL 1 - General Declaration” should be shown
- Upload an image file
- Click on `Check for errors`
- > Error of “The selected file must be a csv or xlsx” should be shown
- Upload a large file (over 4MB)
- Click on `Check for errors`
- > Error of “The file must be smaller than 4MB” should be shown
- Click on `The file must be smaller than 4MB` link in the `Error summary panel`
- > You should be scrolled to the Upload a file question
- Upload a new file
- > The error message should clear
- Delete your auth token
- Click on `Check for errors`
- > You should be redirected to sign in
- Sign in
- > You should directed back to the Upload general declaration FAL 1 page

 

For dev testing 404

- In `FileUploadForm` change the endpoint to `/declaration//upload-fal1`
- Upload a file
- Click `Check for errors`
- > You should be taken to something went wrong page
- Click link
- You should be taken to Your voyages page

 

SUCCESS SCENARIO:

- Upload the FAL 1 template file
- Click `Check for errors`
- > You should be taken to the confirmation of upload page with a success message
- Click `Save and continue`
- > You should be taken to the task details page

----

## Notes

Waiting on a fix to BE which currently returns a 500 error in what we expect to be a success
